crates/turbo-tasks/src/backend.rs

```diff
@@ -99,28 +100,206 @@ impl Display for PersistentTaskType {
     }
 }
 
-impl PersistentTaskType {
-    pub fn shrink_to_fit(&mut self) {
-        match self {
-            Self::Native {
-                fn_type: _,
-                this: _,
-                arg,
-            } => arg.shrink_to_fit(),
-            Self::ResolveNative {
-                fn_type: _,
-                this: _,
-                arg,
-            } => arg.shrink_to_fit(),
-            Self::ResolveTrait {
-                trait_type: _,
-                method_name: _,
-                this: _,
-                arg,
-            } => arg.shrink_to_fit(),
+mod ser {
```

Comment:
Do we have any unit tests that round-trip this serialization logic? If not, it might be nice to have.

Reply:
not yet, I'll add them as part of persistent caching